internal/bminventory/inventory_test.go

```diff
@@ -4653,24 +4653,87 @@ var _ = Describe("TestRegisterCluster", func() {
 		verifyApiError(reply, http.StatusBadRequest)
 	})
 
-	It("NTPSource default value", func() {
-		mockClusterApi.EXPECT().RegisterCluster(ctx, gomock.Any()).Return(nil).Times(1)
-		mockEvents.EXPECT().
-			AddEvent(gomock.Any(), gomock.Any(), nil, models.EventSeverityInfo, gomock.Any(), gomock.Any()).
-			Times(1)
-		mockMetric.EXPECT().ClusterRegistered(gomock.Any(), gomock.Any(), gomock.Any()).Times(1)
-		mockSecretValidator.EXPECT().ValidatePullSecret(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Times(1)
+	Context("NTPSource", func() {
+		It("NTPSource default value", func() {
+			By("Empty", func() {
+				Expect(os.Unsetenv("NTP_DEFAULT_SERVER")).ShouldNot(HaveOccurred())
+				Expect(envconfig.Process("test", &cfg)).ShouldNot(HaveOccurred())
 
-		reply := bm.RegisterCluster(ctx, installer.RegisterClusterParams{
-			NewClusterParams: &models.ClusterCreateParams{
-				Name:             swag.String("some-cluster-name"),
-				OpenshiftVersion: swag.String("4.6"),
-				PullSecret:       swag.String(`{\"auths\":{\"cloud.openshift.com\":{\"auth\":\"dG9rZW46dGVzdAo=\",\"email\":\"coyote@acme.com\"}}}"`),
-			},
+				bm = NewBareMetalInventory(db, getTestLog(), nil, mockClusterApi, cfg, nil, mockEvents,
+					nil, mockMetric, getTestAuthHandler(), nil, nil, mockSecretValidator)
+				Expect(bm.Config.DefaultNTPSource).Should(BeEmpty())
+
+				mockClusterApi.EXPECT().RegisterCluster(ctx, gomock.Any()).Return(nil).Times(1)
+				mockEvents.EXPECT().
+					AddEvent(gomock.Any(), gomock.Any(), nil, models.EventSeverityInfo, gomock.Any(), gomock.Any()).
+					Times(1)
+				mockMetric.EXPECT().ClusterRegistered(gomock.Any(), gomock.Any(), gomock.Any()).Times(1)
+				mockSecretValidator.EXPECT().ValidatePullSecret(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Times(1)
+
+				reply := bm.RegisterCluster(ctx, installer.RegisterClusterParams{
+					NewClusterParams: &models.ClusterCreateParams{
+						Name:             swag.String("some-cluster-name"),
+						OpenshiftVersion: swag.String("4.6"),
+						PullSecret:       swag.String(`{\"auths\":{\"cloud.openshift.com\":{\"auth\":\"dG9rZW46dGVzdAo=\",\"email\":\"coyote@acme.com\"}}}"`),
+					},
+				})
+				Expect(reflect.TypeOf(reply)).Should(Equal(reflect.TypeOf(installer.NewRegisterClusterCreated())))
+				actual := reply.(*installer.RegisterClusterCreated)
+				Expect(actual.Payload.AdditionalNtpSource).To(BeEmpty())
+			})
+
+			By("Nonempty", func() {
+				defaultNtpSource := "clock.redhat.com"
+
+				Expect(os.Setenv("NTP_DEFAULT_SERVER", defaultNtpSource)).ShouldNot(HaveOccurred())
```

Comment:
I wonder if it can potentially affect other tests?
Maybe instead you just set the configuration manually after `Expect(envconfig.Process("test", &cfg)).ShouldNot(HaveOccurred())
` ?

Reply:
It won't affect because it resets the config and the bminventory instance on `BeforeEach`
```golang
	BeforeEach(func() {
		Expect(envconfig.Process("test", &cfg)).ShouldNot(HaveOccurred())
		db = common.PrepareTestDB(dbName)
		ctrl = gomock.NewController(GinkgoT())

		mockClusterApi = cluster.NewMockAPI(ctrl)
		mockEvents = events.NewMockHandler(ctrl)
		mockMetric = metrics.NewMockAPI(ctrl)
		mockSecretValidator = validations.NewMockPullSecretValidator(ctrl)
		bm = NewBareMetalInventory(db, getTestLog(), nil, mockClusterApi, cfg, nil, mockEvents,
			nil, mockMetric, getTestAuthHandler(), nil, nil, mockSecretValidator)
	})
```

Reply:
i'm talking about env, the evn will still exist on other tests no?

Reply:
maybe i wasn't clear:
```go
package testing

import (
	"fmt"
	"os"
	"testing"
	"time"

	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
)

func TestCluster(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "tests")
}

var _ = Describe("test", func() {
	It("env", func() {
		os.Setenv("FOO", "BAR")
	})
})

var _ = Describe("test", func() {
	It("env", func() {
		time.Sleep(time.Second)
		fmt.Println(os.Getenv("FOO"))
	})
})
```